### ISSUE TITLE
Integrate thumb up/down or like/dislike

### ISSUE THREAD

**Issue by jiri-janousek** [tiliado/nuvola-app-amazon-cloud-player#29]
Reticketed from https://github.com/tiliado/nuvola-app-amazon-cloud-player/issues/27#issuecomment-320443118
> Does Amazon contain any rating system like thumb up/down, like/dislike or star rating? If yes, could you look at [track rating feature](https://tiliado.github.io/nuvolaplayer/development/apps/mediaplayer.html#track-rating)? This can wait for release 5.6 though if you wish so.

**Comment by jiri-janousek** [comment]
Hello, @a-m-s. Would it be possible to include this in the next release? Do you need some assistance?

**Comment by a-m-s** [comment]
Sorry, it just got forgotten.

**Comment by a-m-s** [comment]
This is now implemented, but there's no default key binding defined, which probably isn't helpful. Otherwise, the Amazon app seems to treat thumb up as a toggle button, and thumb down as a simple action, which may be confusing for the UI.

Do you have any precedence for thumb bindings?

**Comment by a-m-s** [comment]
No reply, so I'm just going to call this done.

**Comment by jiri-janousek** [comment]
Sorry, I've forgotten to finish and send my reply. I don't have any preference for the thumbs keybinding, users can set whatever they wish.
Regarding that thumbs down behavior: Google Play Music sets the thumbs-down state and then skips to the next song. If you play the playlist again and get to the thumb-down song, you can press thumbs down to remove the thumbs-down state. What does Amazon do?

**Comment by a-m-s** [comment]
The interface shows both thumb up and thumb down buttons, which suggests there could be three states: up, down, neither.

Actually clicking the button suggests that there are really only two states: up, or not. The thumbs down button seems to do the same thing as toggling the thumbs up off.

It does not do anything like skipping the track.

**Comment by jiri-janousek** [comment]
If there are only two states, there is no need for the thumbs down toggle action. You can either remove it or make it a simple action, which should be disabled when the thumbs-up state is toggled off.

**Comment by a-m-s** [comment]
I think I must be going slightly crazy!? :-(

I just went back to check what it was doing, and indeed it works like you describe google music. You hit thumbs down and it appears not to toggle because it goes straight to the next track.

I know it was late and I was editing this with the sound off, but I don't know how I got confused with this? Maybe I'd screwed up the classes and the site wasn't working right?

Anyway, the current implementation is correct. Both thumb up and down are toggles, and the actions work just fine.

... I'm now wondering how many great tracks I'll never hear again because I've accidentally given them thumb down?

Incidentally the pop-up text suggests that the thumb setting only applies to the current "station", and there are no thumbs at all for "My Music" or playlists (only the fake radio stations).
